### PR TITLE
feat(client): implement clientside pagination for profile taken_quizzes and bookmarked_posts

### LINKED ISSUE #757: Write Tests For Snake Case to Title Case Utility Function
# 🧱 Description

## 📋 Acceptance Criteria

- [ ] There are 5 unit tests covering different scenarios for the utility fn `snakeToTitle`

### LINKED ISSUE #773: Toast Message on Forum Actions
# 🧱 Description

Display a success or fail toast notification upon trying to create a post with a photo.
Refer to the old usages of `useToastStore()`

blocked by #739 

## 📋 Acceptance Criteria

- [ ] A toast message appears upon creating a forum question successfully or in case of a failure.

### LINKED ISSUE #792: Enhance Ask Community Default Text
# 🧱 Description

When we ask community about a quiz question or a specific choice, we automatically fill the input fields for the users.
However the current default text for such auto-fill text isn't really helpful.

It was also pointed out in the CM2 and caused some confusion around.

We should change the default text to be more helpful and realistic

CleanShot 2024-12-14 at 05 14 34@2x

CleanShot 2024-12-14 at 05 15 27@2x

CleanShot 2024-12-14 at 05 16 25@2x

## 📋 Acceptance Criteria

- [ ] Ask to Community buttons have more meaningful and realistic auto-fill texts

### LINKED ISSUE #822: Display rating only if there's at least 1 rating
# 🧱 Description

Display the rating only if there's at least 1 rating.

CleanShot 2024-12-14 at 23 30 22@2x

## 📋 Acceptance Criteria

- [ ] No N/A needed on Quizzes

### LINKED ISSUE #824: Client Side Pagination & Load More Button for Profil Items
# 🧱 Description

When the user has too many bookmarked questions and taken quizzes, this will be mess.
Set the initial count to 6 and increment it 6 by 6 as user clicks "Load More" upon both sections

CleanShot 2024-12-15 at 01 36 40@2x

## 📋 Acceptance Criteria

- [x] There's a Client Side Pagination in Profile Items

### PR DESCRIPTION
**Pull request by hks1444**
Closes #824 Closes #822 Closes #792 Closes #773 Closes #757

### CODE REVIEW

File: client/src/tests/snake.test.ts
Comment (umitcan07) [inline client/src/tests/snake.test.ts:1]: bad file name, make it `string.ts` pls

File: client/src/routes/Forum/Forum.data.tsx
Comment (umitcan07) [inline client/src/routes/Forum/Forum.data.tsx:1]: please revert the changes in this file, i have made a major refactor for this one. i simply copy-pasted you changes with the toast already.

File: client/src/components/quiz-card.tsx [hunk 1]
```diff
@@ -81,21 +81,23 @@ export const QuizCard = ({ quiz, onTagClick, quiz_key }: QuizCardProps) => {
...
-                                    score: scoreToInteger(
-                                        quiz.rating.score || 0,
-                                    ),
-                                })}
-                            >
-                                {quiz.rating.score || "N/A"}
-                            </span>
-                        }
-                    ></TooltipAnchor>
+                    {quiz.rating.score && (
```
Comment (umitcan07) [inline client/src/components/quiz-card.tsx:84]: `score` is an integer
if its 0, we will display 0 instead of the component in the paranthesis.

make it 

`quiz.rating.score !== null`

File: client/src/components/bookmarked-forums.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { Button } from "@ariakit/react";
+import { useState } from "react";
+import { ForumQuestion } from "../routes/Forum/Forum.schema";
+import { buttonClass } from "./button";
+import { ForumQuestionCard } from "./forum-card";
+
+const INITIAL_DISPLAY_COUNT = 6;
+const LOAD_MORE_COUNT = 6;
```
Comment (umitcan07) [inline client/src/components/bookmarked-forums.tsx:8]: nice one

File: client/src/tests/snake.test.ts
Comment (hks1444) [reply client/src/tests/snake.test.ts:1]: it should be string.test.ts

**Review by umitcan07 — APPROVED** [review]
